local-queue.js

```diff
@@ -0,0 +1,73 @@
+const { weightSrvRecords } = require("ioredis/built/cluster/util")
+
+const queues = {}
+
+// Get or create queue
+function getOrCreateQueue(queueName) {
+  q = null
+  if (!(queueName in queues)) {
+    q = []
+    queues[queueName] = q 
+  } else {
+    q = queues[queueName]
+  }
+  return q
+}
+
+// Enqueue an item to the queue
+async function push(queueName, item) {
+  q = getOrCreateQueue(queueName)
+  // Check if the element already exists in the list
+  if (!(item in q)) {
+    q.push(item)
+  }
+  q.push(item)
+}
+
+// Enqueue an item to the queue and get the queue
+async function pushAndGetQueue(queueName, item) {
```

Comment:
Why are all these functions `async`? I don't think there is any need for that (and I think it is actually worse for performance, since it wraps its return value into a `Promise`).

Reply:
Since I'm dropping in the local-queue.js instead of redis-queue.js I kept the function signatures the same. I'll see if it can be better.

Reply:
switched to non aync functions wince we are not focusing on redis at the momnet